pkg/operator/controller/ingress/controller.go

```diff
@@ -421,17 +421,23 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig
 		}
 	case operatorv1.NodePortServiceStrategyType:
 		// Update if PROXY protocol is turned on or off.
+		if ic.Status.EndpointPublishingStrategy.NodePort == nil {
+			ic.Status.EndpointPublishingStrategy.NodePort = &operatorv1.NodePortStrategy{}
+		}
 		statusNP := ic.Status.EndpointPublishingStrategy.NodePort
 		specNP := effectiveStrategy.NodePort
-		if specNP != nil && statusNP != nil && specNP.Protocol != statusNP.Protocol {
+		if specNP != nil && specNP.Protocol != statusNP.Protocol {
```

Comment:
I still have a question about this - if someone sets `ic.Spec.EndpointPublishingStrategy.NodePort` to nil (or removes it), then it won't update ic.Status nor return true here, will it?

Reply:
Line 375 initializes `ic.Spec.EndpointPublishingStrategy.NodePort` if it is nil.  In fact, for that reason, we probably don't need the `specNP != nil` check anyway.  
https://github.com/openshift/cluster-ingress-operator/blob/8b258e529c1031e193672df61c9bc020c3438953/pkg/operator/controller/ingress/controller.go#L351
https://github.com/openshift/cluster-ingress-operator/blob/8b258e529c1031e193672df61c9bc020c3438953/pkg/operator/controller/ingress/controller.go#L366
https://github.com/openshift/cluster-ingress-operator/blob/8b258e529c1031e193672df61c9bc020c3438953/pkg/operator/controller/ingress/controller.go#L373-L375

Reply:
Okay.  That had a code smell and I guess this explains it.